### PR TITLE
validation script + cost in the roadmap infrastructure

### PR DESCRIPTION
**Pull request by yammineramy**

### CODE REVIEW

File: src/ompl/tools/thunder/src/SPARSdb.cpp [hunk 1]
```diff
@@ -1682,7 +1680,12 @@ void ompl::geometric::SPARSdb::connectGuards(Vertex v, Vertex vp)
     Edge e = (boost::add_edge(v, vp, g_)).first;
 
     // Add associated properties to the edge
-    edgeWeightProperty_[e] = costFunction(v, vp);  // TODO: use this value with astar
+    if (edge_weight.has_value()) {
+        edgeWeightProperty_[e] = edge_weight.value().value();
```
Comment (brook-stevens) [inline src/ompl/tools/thunder/src/SPARSdb.cpp:1684]: Can the second value() be null here?

Comment (yammineramy) [reply src/ompl/tools/thunder/src/SPARSdb.cpp:1684]: No I dont think so, the constructor of the ob::Cost class sets the value to zero by default

File: src/ompl/base/PlannerData.h [hunk 2]
```diff
@@ -340,6 +341,8 @@ namespace ompl
             /// out-variable \e weight. Otherwise, this function
             /// returns false.
             bool getEdgeWeight(unsigned int v1, unsigned int v2, Cost *weight) const;
+            
+            std::optional<ompl::base::Cost> getEdgeWeightOrNull(unsigned int v1, unsigned int v2) const;
```
Comment (brook-stevens) [inline src/ompl/base/PlannerData.h:345]: getEdgeWeightOrNull -> getEdgeWeightIfExists

Comment (brook-stevens) [reply src/ompl/base/PlannerData.h:345]: Add comment as is standard per lib

**Pull request merged**